### PR TITLE
bugfix: LuaJIT sets currentline to -1 if it is unavailable.

### PR DESCRIPTION
**Pull request by spacewander**
I hereby granted the copyright of the changes in this pull request
to the authors of this lua-nginx-module project.

### CODE REVIEW

File: src/ngx_http_lua_log.c [hunk 1]
```diff
@@ -194,7 +194,7 @@ log_wrapper(ngx_log_t *log, const char *ident, ngx_uint_t level,
     *p++ = ':';
 
     p = ngx_snprintf(p, NGX_INT_T_LEN, "%d",
-                     ar.currentline ? ar.currentline : ar.linedefined);
+                     ar.currentline != -1 ? ar.currentline : ar.linedefined);
```
Comment (agentzh) [inline src/ngx_http_lua_log.c]: Better be `>= 0` instead.

**Comment by spacewander** [comment]
@agentzh 
Updated.

File: src/ngx_http_lua_log.c [hunk 2]
```diff
@@ -194,7 +194,7 @@ log_wrapper(ngx_log_t *log, const char *ident, ngx_uint_t level,
     *p++ = ':';
 
     p = ngx_snprintf(p, NGX_INT_T_LEN, "%d",
-                     ar.currentline ? ar.currentline : ar.linedefined);
+                     ar.currentline >= 0 ? ar.currentline : ar.linedefined);
```
Comment (agentzh) [inline src/ngx_http_lua_log.c]: Sorry, I meant 0 line number makes little sense, so we should use `>` here instead? Just like the previous behavior?

Comment (spacewander) [reply src/ngx_http_lua_log.c]: @agentzh 
Updated again.

**Pull request merged**